cmd/thanos/tools_delete.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cespare/xxhash"
+	"github.com/go-kit/kit/log"
+	"github.com/oklog/run"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/thanos-io/thanos/pkg/model"
+	"github.com/thanos-io/thanos/pkg/runutil"
+
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/objstore/client"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+)
+
+type Tombstone struct {
```

Comment:
Probably we need Thanos `tombstone` package (:

Reply:
Yes